File: editor/src/main/java/com/libreshockwave/editor/score/PlaybackHead.java
package com.libreshockwave.editor.score;

/**
 * Tracks the playback head position in the score timeline.
 */
public class PlaybackHead {

    private int frame = 1;

    public int getFrame() {
        return frame;
    }

    public void setFrame(int frame) {
        this.frame = Math.max(1, frame);
    }
}

File: cpp/src/editor/score/PlaybackHead.cpp
#include "libreshockwave/editor/score/PlaybackHead.hpp"

#include <algorithm>

namespace libreshockwave::editor::score {

int PlaybackHead::frame() const {
    return frame_;
}

void PlaybackHead::setFrame(int frame) {
    frame_ = std::max(1, frame);
}

} // namespace libreshockwave::editor::score
